src/shogun/lib/exception/ShogunException.h

```diff
@@ -33,21 +32,11 @@ namespace shogun
 		 */
 		explicit ShogunException(const char* what_arg);
 
-		/** copy constructor
-		 *
-		 * @param orig source object
-		 */
-		ShogunException(const ShogunException& orig);
-
-		/** destructor
-		 */
-		virtual ~ShogunException();
-
-		virtual const char* what() const noexcept override;
+		const char* what() const noexcept override;
 
 	private:
-		/** exception string */
-		std::string msg;
+		/** exception object */
+		std::runtime_error msg;
```

Comment:
we could just inherit from `std::runtime_error` or? i mean that `ShogunException: public std::runtime_error` and then we are good :)